File: src/Repository/IRepository.java
package Repository;

import Model.ProgramState;

/**
 * Created by vladc on 22.10.2016.
 */
public interface IRepository {
    ProgramState getCurrentProgram();
    void add(ProgramState state);
    void logProgramStateExec();
    void logEndOfExecution();
    void serialize();
    ProgramState deserialize();
}

File: Toy-Language-Interpreter_C#/Toy-Language-Interpreter/Repository/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.Model;

namespace Toy_Language_Interpreter.Repository
{
    interface IRepository
    {
        ProgramState GetCurrentProgram();
        void Add(ProgramState state);
        void LogProgramStateExec(ProgramState state);
    }
}
